src/components/shared/swipeable-modal/SwipeableModal.test.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import "hammerjs";
+
+import SwipeableModal from "./SwipeableModal";
+
+test("Should render depends on open prop ", async () => {
+  const { rerender } = render(
+    <SwipeableModal open={true}>Some content</SwipeableModal>
+  );
+  expect(screen.getByRole("dialog")).toBeInTheDocument();
+
+  rerender(<SwipeableModal open={false}>some content</SwipeableModal>);
+  expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
+});
+
+test("Should call onClose when swipe modal", async () => {
+  const touchstart = {
+    bubbles: true,
+    clientX: 0,
+    clientY: 0,
+  };
+  const touchSwipe = {
+    bubbles: true,
+    clientX: 600,
+    clientY: 0,
+  };
+  const onClose = jest.fn();
+
+  render(
+    <SwipeableModal open={true} onClose={onClose}>
+      <button>Some content</button>
+    </SwipeableModal>
+  );
+  const modal = screen.getByRole("dialog");
```

Comment:
It's not great, but if you mock the `clientWidth` value of the modal, it removes the warning:

```suggestion
  const modal = screen.getByRole("dialog");
  jest.spyOn(modal, "clientWidth", "get").mockImplementation(() => 100);

```